GitUI/Editor/CustomSyntaxStrategyManager.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ICSharpCode.TextEditor.Document;
+
+namespace GitUI.Editor
+{
+    // simple manager which return highighting strategy implemented in current assembly
+    class CustomSyntaxStrategyManager
```

Comment:
looking at the code this doesn't feel like a manager class but like a factory
so personally I would call this `SyntaxStrategyFactory`